fv/ipip_test.go

```diff
@@ -0,0 +1,178 @@
+		defaultProfile := api.NewProfile()
+		defaultProfile.Name = "default"
+		defaultProfile.Spec.LabelsToApply = map[string]string{"default": ""}
+		defaultProfile.Spec.Egress = []api.Rule{{Action: api.Allow}}
+		defaultProfile.Spec.Ingress = []api.Rule{{Action: api.Allow}}
+		_, err := client.Profiles().Create(utils.Ctx, defaultProfile, utils.NoOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait until the tunl0 device appears; it is created when felix inserts the ipip module
+		// into the kernel.
+		Eventually(func() error {
+			links, err := netlink.LinkList()
+			if err != nil {
+				return err
+			}
+			for _, link := range links {
+				if link.Attrs().Name == "tunl0" {
+					return nil
+				}
+			}
+			return errors.New("tunl0 wasn't auto-created")
+		}).Should(BeNil())
+
+		// Create workloads, using that profile.  One on each "host".
+		for ii := range w {
+			wIP := fmt.Sprintf("10.65.%d.2", ii)
+			wIface := fmt.Sprintf("cali1%d", ii)
+			wName := fmt.Sprintf("w%d", ii)
+			w[ii] = workload.Run(felixes[ii], wName, wIface, wIP, "8055", "tcp")
+			w[ii].Configure(client)
+
+			hostW[ii] = workload.Run(felixes[ii], fmt.Sprintf("host%d", ii), "", felixes[ii].IP, "8055", "tcp")
+		}
+
+		cc = &workload.ConnectivityChecker{}
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			for _, felix := range felixes {
+				felix.Exec("iptables-save", "-c")
+				felix.Exec("ip", "r")
+			}
+		}
+
+		for _, wl := range w {
+			wl.Stop()
+		}
+		for _, wl := range hostW {
+			wl.Stop()
+		}
+		for _, felix := range felixes {
+			felix.Stop()
+		}
+
+		if CurrentGinkgoTestDescription().Failed {
+			etcd.Exec("etcdctl", "ls", "--recursive", "/")
+		}
+		etcd.Stop()
+	})
+
+	It("should have workload to workload connectivity", func() {
+		cc.ExpectSome(w[0], w[1])
+		cc.ExpectSome(w[1], w[0])
+		cc.CheckConnectivity()
+	})
+
+	It("should have host to workload connectivity", func() {
+		cc.ExpectSome(felixes[0], w[1])
+		cc.ExpectSome(felixes[0], w[0])
+		cc.CheckConnectivity()
+	})
+
+	It("should have host to host connectivity", func() {
+		cc.ExpectSome(felixes[0], hostW[1])
+		cc.ExpectSome(felixes[1], hostW[0])
+		cc.CheckConnectivity()
+	})
+
+	Context("with host protection policy in place", func() {
+		BeforeEach(func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+			defer cancel()
+
+			for _, f := range felixes {
+				hep := api.NewHostEndpoint()
+				hep.Name = "eth0-" + f.Name
+				hep.Spec.Node = f.Hostname
+				hep.Spec.ExpectedIPs = []string{f.IP}
+				_, err := client.HostEndpoints().Create(ctx, hep, options.SetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
+
+		It("should have workload connectivity but not host connectivity", func() {
+			// Host endpoints block host-host traffic.
+			cc.ExpectNone(felixes[0], hostW[1])
+			cc.ExpectNone(felixes[1], hostW[0])
+			// But the rules to allow IPIP between our hosts let the workload traffic through.
```

Comment:
Add "and the lack of applyOnForward"?

Reply:
Discussed offline and improved surrounding comment.